### ISSUE TITLE
Getting Err:504 even in the Examples.ods

### ISSUE THREAD

**Issue by endrepalinkas** [walkjivefly/LOC-Extension#4]
Thank you for taking the time to report an issue. To help me help you please specify

**Comment by Wicked0ne** [comment]
I'm also experiencing this issue. Any solutions yet?

**Comment by endrep0** [comment]
Thanks for looking into it.

I downloaded the new Example.ods from today, and it's still giving the same Err:504 messages.

LOC-Extension version: 0.2.3

**Comment by walkjivefly** [comment]
That's weird, and I'm sorry you're (still) having a problem, @endrep0. I still don't have access to a Windows machine to test/fix it on. If you open one of the cells with the Err:504 message, what is the value shown in the formula bar? Is it something like 
`=CCXT(B8,CONCAT(A8,"/usd"),"price_btc")*C8`
or is the function name something long and complicated? And which version of LibreOffice are you using?

**Comment by endrep0** [comment]
LibreOffice 5.3.7.2 (x64) (might have upgraded once since the original issue was posted)

Yes - the formulas in the formula bars look just like that.  
Let me copy-paste a few examples, all result in Err:504:
`=GETPOLONIEX(E3;F3)`
`=CCXT(F2;F3;F4)`
`=CCXT("coinmarketcap";CONCAT(A6;"/usd");"price_btc")` etc.

Thank you for trying to fix it :)
Maybe you could test it on something like this?
https://developer.microsoft.com/en-us/windows/downloads/virtual-machines
Just a guess, I'm not 100% sure if they can be used for stuff like this :)

**Comment by walkjivefly** [comment]
You have semicolons in the formulae? I have commas, but semicolons work for me too. What language settings do you have?
I've tested on 5.3.4/5 and 5.4 but not 5.3.7. I'm downloading it to try.
I did the limited testing with Windows 7 using a VirtualBox VM and the MS developer download but don't have access to it at present.

**Comment by walkjivefly** [comment]
You have semicolons in the formulae? I have commas, but semicolons work for me too. What language settings do you have?
I've tested on 5.3.4/5 and 5.4 but not 5.3.7. I'm downloading it to try.
I did the limited testing with Windows 7 using a VirtualBox VM and the MS developer download but don't have access to it at present.

**Comment by endrep0** [comment]
Yes semicolons.
If I open the file you gave it's all semicolons already, I'm guessing it's using the correct format for my locale. In other functions like `=MAX` I also have to use semicolons.

The LibreOffice is in English, and the Locale is hu-HU (hu_HU).

**Comment by walkjivefly** [comment]
Wow, with 5.3.7.2 I can't open ANY sheets from 5.3.5, including Example.ods
I'm going to uninstall and re-install and try again...

**Comment by Dryparn** [comment]
Yeah, i get exceptions with coinmarketcap on linux(alpha4) and error 504 on pretty much everything in windows(alpha3w).

**Comment by vadirthedark** [comment]
OS is Ubuntu 18.04; LOC v1.6.0.7 on a Linux system. Having the same issues as many of the above (exceptions encountered, #VALUE). When I 'Reload' I get a box asking if I want to 'Delete Change?'. Is there a place with more details on working with your template (cctx, get, etc.)? Thanks

**Comment by walkjivefly** [comment]
@vadirthedark I take it that version number is Ubuntu's way of saying LibreOffice 6.0.7? I'm using the extension with 6.0.1.1 on MX 17.1 linux (somewhat Ubuntu 18-like). I haven't reproduced any of the Err:504's people have reported so I don't know what causes them. 

The example.ods is pretty much all the doc I wrote. The ccxt() function takes 3 parameters; an exchange, a ticker and a function, eg: 
```
=CCXT("bitfinex","btc/usd","last")
```
or
```
=CCXT("coinmarketcap","eth/usd","price_btc")
```
If you change any values in your sheet you should save it before using File/Reload to force the extension to refresh an exchange with cached results by using
```
=ccxt("bitmex","reload")
```
Only a handful of exchanges provide cached results to the extension. Most ccxt() calls with Coinmarketcap as the exchange fail since they switched their API last December. I have no plans to try to update the extension in the near future.

**Comment by vadirthedark** [comment]
I appreciate the additional information and will continue on.
One question-are each line to be placed into a separate cell?

Thanks again

**Comment by walkjivefly** [comment]
I don't understand the question. Here's another example of how I use it:
![Screenshot_2019-03-10_11-52-23](https://user-images.githubusercontent.com/17745270/54080846-b9b7dc80-432b-11e9-8751-d4000b998734.png)

Column A: ticker symbol
Column B: exchange name
Column C: use exchange (=1) or aggregator (Coinmarketcap) (=0) price
Column D: price source to use
Column E: number of coins
Column F: price paid (bought with BTC)
Column G: price paid (bought with ETH)
Column H: current value on the named exchange or on aggregator
Column I: change in %
Column J: profit/loss in BTC
